### PR TITLE
New `some` function for collections

### PR DESCRIPTION
**Pull request by allejo**
Introduce an equivalent of JS' `Array.some` function, which returns true if at least one item in a collection evaluates to true with respect to the given callback.

```php
// true
$hasEvenNumber = __::some([1, 3, 5, 10, 7, 9], static function ($item, $key, $collection) {
    return $item % 2 === 0;
});
```

I don't _think_ this exists natively in PHP nor in this library, but please do let me know if it does.

### CODE REVIEW

File: src/__/collections/some.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * @template V
+ * @template K
+ *
+ * @param iterable|\stdClass $collection
+ * @param \Closure           $callback
+ *
+ * @return bool
+ */
+function some($collection, \Closure $callback)
```
Comment (MonsieurV) [inline src/__/collections/some.php:35]: We may add `__::identity` as the default $callback, so we can just do `__:some` on a collection of booleans (or "trueables") without having to write/pass the callback.

This is the case in Lodash [`some`](https://lodash.com/docs/4.17.15#some), Elixir [`any?`](https://hexdocs.pm/elixir/1.12/Enum.html#any?/1) (but not in Clojure [`some`](https://clojuredocs.org/clojure.core/some), but there users are more familiar with functional composition).

If this is accepted `__:every` must also be updated.

**Review by MonsieurV — CHANGES_REQUESTED** [review]
Right! We had `__:every` but were missing this some/any

File: src/__/collections/some.php (hunk 1 above)
Comment (allejo) [reply src/__/collections/some.php]: I can 100% get behind defaulting the callbacks to `__::identify()` like other libraries. Pushed up those changes 👍

**Review by MonsieurV — APPROVED** [review]
Parfait :)

**Pull request merged**